exporter/collector/metrics.go

```diff
@@ -190,7 +190,17 @@ func (me *MetricsExporter) PushMetrics(ctx context.Context, m pmetric.Metrics) e
 	// they need to be sent to. Each project's timeseries are sent in a
 	// separate request later.
 	pendingTimeSeries := map[string][]*monitoringpb.TimeSeries{}
-	rms := m.ResourceMetrics()
+
+	rms := pmetric.NewResourceMetricsSlice()
+	m.ResourceMetrics().CopyTo(rms)
```

Comment:
One downside of this approach we should avoid if possible: This performs a copy even if `me.cfg.MetricConfig.ExtraMetrics == nil`, which will be the case in our normal exporter.  We should avoid this case if possible. Even if you don't accept my suggestion below.

Reply:
Moved this around to avoid the copy if `ExtraMetrics == nil`, ptal